library/psa_crypto.c

```diff
@@ -5728,8 +5736,8 @@ static psa_status_t psa_generate_derived_key_internal(
         return PSA_ERROR_INVALID_ARGUMENT;
     }
 
-#if defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_KEY_PAIR) || \
-    defined(MBEDTLS_PSA_BUILTIN_KEY_TYPE_ECC_PUBLIC_KEY) || \
+#if defined(PSA_WANT_KEY_TYPE_ECC_KEY_PAIR) || \
+    defined(PSA_WANT_KEY_TYPE_ECC_PUBLIC_KEY) || \
     defined(MBEDTLS_PSA_BUILTIN_ALG_ECDSA) || \
     defined(MBEDTLS_PSA_BUILTIN_ALG_DETERMINISTIC_ECDSA) || \
     defined(MBEDTLS_PSA_BUILTIN_ALG_ECDH)
```

Comment:
This is another minor thing that is not blocking PR, but I was wondering: now that we changed from `MBEDTLS_PSA_BUILTIN_KEY_` to `PSA_WANT_KEY_TYPE`, can't we also drop the following `MBEDTLS_PSA_BUILTIN_ALG` guards? 
To me it looks like the latter (`MBEDTLS_PSA_BUILTIN_ALG`) are a subgroup of the former (`PSA_WANT_KEY_TYPE`), but probably I'm missing something here, isn't it?

Reply:
I have to say, I'm not entirely sure why the `BUILTIN_ALG` flags were included here. That's probably something we want to look at, at some point.